### DISCUSSION TITLE
Display Popover onFocus of Input Field

### DISCUSSION THREAD

**Discussion by ghost** [twilio-labs/paste#1342] [Q&A] [answered]
As per design, I wanted to show popover when user focus on particular field. 
Actual usage of it is on reset password page, we wanted to show password guideline when user try to enter new password. Since we have so many guidelines, it can't be something simple like tooltip. 

According to paste doc, Only button click can display popover. Can we have primitive/state for popover also, so that I can customise this?

**Comment by serifluous** [comment, marked as answer]
Hey @Kajal-Joshi , thanks for the question! Generally, we suggest that [password guidelines are shown using Help Text that appears below the form field](https://paste.twilio.design/components/input#labels-and-help-text), rather than in tooltips or popovers, to help prevent errors up-front as much as possible.

If that solution doesn't work for your case, please let us know!

**Reply by ghost** [reply]
@serifluous 
 According to design we wanted to show something like below image.

When user start typing , displaying these many guideline as helptext would not be good UX. Plus we first wanted to give minimal guidelines and when user clicks on "show more", then only display whole list of guidelines.

![image](https://user-images.githubusercontent.com/72342162/114008760-2e013900-9880-11eb-81d5-37510ca90965.png)

![image](https://user-images.githubusercontent.com/72342162/113962587-9b43a880-9845-11eb-9f2f-398987821a18.png)

**Reply by pguptaDS** [reply]
@serifluous Just to add - we considered placing a helper text but as you can see we have to display "password guidelines" which makes this complicated. We are open to suggestions if you have any!

**Reply by serifluous** [reply]
That helps, thanks! I'd recommend:

- Placing the password requirements ("At least 14 characters...") as help text under the form field since that's the only information required to fill out the field.
- I still think there's an opportunity to show the password guidelines directly on the page, rather than a popover. Directionally, something like this might help, especially if you want to keep the layout similar: https://www.figma.com/file/SpeLpBoqN2vZv8KWcSn0J1/Feedback?node-id=303%3A6524

If you do want to use a popover, the most common composition we've seen in the past is adding a suffix to the form field, and putting an information icon button in it. See the second example in this section: https://paste.twilio.design/components/input/#input-with-add-ons-prefix-suffix-text-or-icons (the "Message Title" input).

**Reply by pguptaDS** [reply]
Thanks @serifluous for your guidance! Kajal and I will try to incorporate your feedback. I am leaning towards this but will have to discuss with Kajal on the implementation: https://share.goabstract.com/2b5d7714-3259-4602-a0f8-90a0bb7a2f7e

**Reply by serifluous** [reply]
Sounds good. Thanks for following up!
